### ISSUE TITLE
Don't touch .bashrc if the tmc executable is already in $PATH

### ISSUE THREAD

**Issue by nygrenh** [testmycode/tmc-cli#254]
This occurs for example when tmc-cli is installed from npm.

One problem with this is that your autocomplete solution seems to depend on the `.bashrc` modification. This could be fixed by installing the appropriate autocompletion files globally (for example to `/etc/bash_completion.d`) when installing and placing the the executable directly to the $PATH

**Comment by jclc** [comment]
Note: also don't add if there is an alias called "tmc"

Edit: Actually, this might be impossible to check since aliases aren't usually initialised in scripts.

**Comment by Salmela** [comment]
We should export variable so that we can easily detect if the tmc-cli is already installed in scripts.

The problem with aliases is that they aren't inherited by sub-shells.

**Comment by Salmela** [comment]
The tmc launcher won't do anything if the tmc auto-completion file exists (`~/.tmc-autocomplete.sh`). It's location can be overridden with `$TMC_AUTOCOMPLETE_FILE` global variable

**Comment by nygrenh** [comment]
@Salmela Expecting package maintainers to set arbitrary environment variables seems unreasonable to me.

**Comment by Salmela** [comment]
Good call. I just assumed that package manager would put the autocompletion script into same place as the tmc launcher script, which wouldn't work very well because the default location was in the home directory.

I now changed it so that if the tmc is in read-only directory then it won't try to create the tmc autocompletion script.
